bundles/org.eclipse.swt/Eclipse SWT/win32/org/eclipse/swt/graphics/Image.java

```diff
@@ -758,11 +771,10 @@ boolean refreshImageForZoom () {
 			refreshed = true;
 			setCurrentDeviceZoom(deviceZoomLevel);
 		}
-	} else {
+	} else if (this.dataAtBaseZoom != null) { // Resizing is only possible with a cached base image
```

Comment:
There are 2 constructors that do **not** initialize `dataAtBaseZoom` (you can see them by marking the field as `final`). Have those 2 cases been tested?

Reply:
Yes, but I will have a look at possible automated tests for it

Reply:
I had a look a the existing tests and would say, that the `test_getImageData_XXX `Tests in `Test_org_eclipse_swt_graphics_Image `cover most of it. I added one test for an additional constructor.

Reply:
Hm, those tests don't walk into the changed `else`-block and neither does the new test.

I also debugged all tests in the package and none of them reach the `else` (neither in your branch nor in the `master` branch). Alos, lots of them fail when I run them locally, maybe I'm doing something wrong? 

Could you please double check that:
1. Any test is reaching the `else`-block before applying your PR and
2. The test(s) that reached the `else` clock in `master` also pass in your branch?

This would at least mean that adding the extra condition to the else didn't break anything (or at least increase the chances that nothing broke because of the change).

Reply:
Just to summarize our discussion: as this behaviour is windows specific, I added a test in `ImageWin32Tests`, that is running into this block. In the current master, I didn't find a proper and real scenario, where I could run into this block.